File: src/pattern/bridge/java/StringDisplayImpl.java
package pattern.bridge.java;

public class StringDisplayImpl extends DisplayImpl {

    public StringDisplayImpl(String str){
        super(str);
    }
    @Override
    public void rawOpen() {
        System.out.println("==========");
    }

    @Override
    public void rawPrint() {
        System.out.println("+ "+this.str+" +");

    }

    @Override
    public void rawClose() {
        System.out.println("=========");
    }
}

File: src/pattern/bridge/kotlin/StringDisplayImpl.kt
package pattern.bridge.kotlin

class StringDisplayImpl(var str: String): DisplayImpl() {


    override fun rawOpen() {
        println("===========")
    }

    override fun rawPrint() {
        println(this.str)
    }

    override fun rawClose() {
        println("===========")
    }


}